### ISSUE TITLE
Feature request: safety valve to prevent dead players' souls from being trapped in problematic maps

### ISSUE THREAD

**Issue by SolraBizna** [Aleph-One-Marathon/alephone#457]
When a player is killed, they can press the action key to respawn or reload from save—but only once their velocity has settled to zero. There are certain situations a player can die in which will prevent their velocity from ever settling down, such as certain "bouncy" map geometry, or platforms with exactly the right parameters. While these maps ought to be fixed, it would be infeasible to track down every single edge case in Marathon's physics, and it would be nice to prevent this entire class of softlock instead.

The solution that comes to my mind is allowing the player to respawn if it's been 15 seconds since they died, even if their velocity hasn't settled. The main obvious downside of this that I can see is that we then have to figure out where to put their death drop items. (Fortunately, there are already lots of other cases where we get weird about death drop items. :P)

**Comment by aaronfreed** [comment]
My immediate inclination is to suggest putting their death drop items at exactly the floor of whatever polygon their corpse is in at respawn time. This would be slightly weird in some cases, but any other option I can think of would be much weirder.

Since this would remove one of many reasons I’ve needed to write stupid Lua scripts that I never felt entirely good about, I’m naturally in full support of this idea. If need be, I’d be fine with some sort of parameter to enable or disable it if we’re worried about breaking scripts that rely on preventing players from respawning.

**Comment by SolraBizna** [comment]
Scripts can prevent players from respawning by blocking the action key press.

**Comment by aaronfreed** [comment]
Good point – people can just update scripts if they break, and now that we have map-specific plugins, it doesn’t even matter if map authors refuse to update their maps and refuse to offer permission to anyone else to do so either.
